### ISSUE TITLE
sql: Add ability to load .sql databases

### ISSUE THREAD

**Issue by llelik8** [tarantool/tarantool#3917]
There is a sample database obtained from the sqlite3 dump. We need a simple way to load this kind of files to SQL.

**Comment by zilveer** [comment]
I cannot see this feature of importing sql files in milestonr 2.3 ?

Any update regarding this would be appreciated.

**Comment by zilveer** [comment]
Looking forward to this import .sql feature! :)

**Comment by romanhabibov** [comment]
@Totktonada

1. PostgreSQL
`psql -h hostname -d databasename -U username -f file.sql`

2. MySQL
`mysql -u username -p database_name < file.sql `

3. MongoDB
`mongoimport --numInsertionWorkers N`

Do we need to do using tarantoolctl?

**Comment by pgulutzan** [comment]
Judging by Chinook's SQLite .sql script file https://raw.githubusercontent.com/lerocha/chinook-database/master/ChinookDatabase/DataSources/Chinook_Sqlite.sql the difficulties for Tarantool seem to be: knowing where a multi-line statement ends, deciding what to do with errors e.g. ignore or ask the user, stripping []s around identifiers and replacing with ""s, replacing SQLite data types NVARCHAR, NUMERIC, DATETIME with Tarantool types.

**Comment by Totktonada** [comment]
It works in mysql way already with `tarantoolctl enter`:

```
$ cat test.sql 
CREATE TABLE s(id INTEGER PRIMARY KEY, value VARCHAR(100))
INSERT INTO s VALUES (1, 'one')
$ tarantoolctl enter tarantool_example --language sql <test.sql
connected to unix/:/run/tarantool/tarantool_example.control
unix/:/run/tarantool/tarantool_example.control> CREATE TABLE s(id INTEGER PRIMARY KEY, value VARCHAR(100))
---
- row_count: 1
...

unix/:/run/tarantool/tarantool_example.control> INSERT INTO s VALUES (1, 'one')
---
- row_count: 1
...
```

However `tarantoolctl connect` does not follow this option.

**Comment by Totktonada** [comment]
@pgulutzan Sorry, missed your comment.

> knowing where a multi-line statement ends

If I understood correctly, then #4317 should help.

> deciding what to do with errors e.g. ignore or ask the user

Good point. I have no answer for now.

> stripping []s around identifiers and replacing with ""s, replacing SQLite data types NVARCHAR, NUMERIC, DATETIME with Tarantool types.

It looks as a task for a separate tool that will transform another database dialect into tarantool's one.

**Comment by pgulutzan** [comment]
I don't think that putting \s at the ends of lines, as issue#4317 suggests, is good for this situation. It seems to me that some person would have to look at the statements and decide where to put the \s, so it would not be automatic.
On the other hand, this is feasible:
() expect that statements end with ; semicolon
() include a simple tokenizer
() find ; semicolons outside comment or string-literal or begin ... end
There is an alternative, looking for a verb that begins a statement, but there are so many special cases (e.g. ON DELETE SET NULL, WITH ... SELECT, GRANT INSERT, CREATE TRIGGER ... BEGIN ... DROP ... END) that it is not worth trying without a recognizer.

**Comment by romanhabibov** [comment]
Now, it is difficult to implement the display of error line number correctly. Most likely, it will be easier to do after the implementation of stack diagnostics #1148.
